src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameOfT.cs

```diff
@@ -144,6 +135,8 @@ public class Frame<TContext> : Frame
             }
             finally
             {
+                // Error occurred, do not return components to pool
+                ResetComponents(poolingPermitted: false);
```

Comment:
OK. I realize this isn't much worse than it was before since we were already reusing headers between requests on the same connection. I still think having shared configuration for reusing stream/header objects would be nice since in principle it's pretty similar.

 My point still stands about an error not necessarily having occurred here even if the headers are initialized. We have a couple places where we do:

    if (SocketInput.RemoteIntakeFin)
    {
        return;
    }